Fetching data and rendering on screen

DariaKryvos opened the pull request:
## What this request does

- fetching data from json file
- rendering data on the  screen
- added styling
- added icons from react-icons
- links to website
 
![Screen Shot 2022-01-18 at 4 55 43 PM](https://user-images.githubusercontent.com/47646986/149974269-ee41ef89-32b1-4ac0-ab90-fc42f5245270.png)


screen

DariaKryvos committed:
Fetching data and rendering on screen

Reinoptland commented:
Hi @DariaKryv 

Here is video with my feedback:

https://share.getcloudapp.com/5zunXYz9

- Remove floating semi colon
- Add icons for clarity
- Add styling and use the screen real estate better on bigger screens
- Align text to the left (easier to read)
- Make the link to the website a link that opens in a new tab
- Remove all logs
- Separate a company card component and pass data via props
- When adding a new dependency make sure to mention it in your PR summary

Reinoptland commented on src/Pages/HomePage.js:
```diff
@@ -0,0 +1,34 @@
+
+export default function HomePage() {
+  const [agencies, setAgencies] = useState([]);
+  console.log("Show Agency", agencies);
```
Remove all logs when you want to merge

Reinoptland commented on src/Pages/HomePage.js:
```diff
@@ -0,0 +1,34 @@
+      <h1>FNDR</h1>;
+      {agencies.map((agency) => {
+        return (
+          <div key={agency.id}>
```
This can be a component

Reinoptland commented:
@jodelajo it's also good for you to look at this PR and this review

DariaKryvos committed:
add changes: styling, removed loges

DariaKryvos committed:
add changes: styling, removed loges

Reinoptland commented:
Screenshot looks cool! I'll review this ASAP! @DariaKryv

Reinoptland commented on src/Pages/HomePage.css:
```diff
@@ -0,0 +1,61 @@
+  margin: 10px;
+}
+.agency-card-container {
+  display: grid;
```
Hey why did choose grid in this case?

Reinoptland commented on src/Pages/HomePage.css:
```diff
@@ -0,0 +1,61 @@
+  border-bottom: 1px solid rgb(119, 118, 118);
+}
+/* .card-container-border {
+  border: 1px solid grey;
```
Try to remove any commented code when you PR 😊

Reinoptland commented on src/Pages/HomePage.js:
```diff
@@ -0,0 +1,68 @@
+          <div className="block">
+            <div className="agency-card-container">
+              <div className="company-logo">
+                <img src={agency.eguideImageSrc} />
```
This needs an alt. Try to check warnings before you PR

Here is some context an the alt attribute:
https://blog.hubspot.com/marketing/image-alt-text

Reinoptland reviewed (commented):
Let's add pagination in the iteration, because on mobile on slow 3g the page takes 80+ seconds to load completely

Reinoptland commented on src/Pages/HomePage.js:
```diff
@@ -0,0 +1,68 @@
+                      <span>
+                        <MdWebAsset />
+                      </span>
+                      <a href={agency.website}>web site</a>
```
This link works, but it's hard to spot it's a link because it lacks color. 

Also it opens the site in the same tab, make it open in a new tab, that way they don't need a page reload when they click back. 

here is how to do that:

https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_link_target

Reinoptland commented on src/Pages/HomePage.css:
```diff
@@ -0,0 +1,61 @@
+.company-info h5 {
+  margin-bottom: 5px;
+}
+span {
```
Keep in mind that you're styling ALL span's when you do this. 

So when someone else uses a span in the rest of the website it will be styled like this. 

That means that person will need to change this and a class most likely. 

You can use tag selector sometimes but only for defining "default" styles for the entire website. The use span is usually very situational, so default don't make that much sense.

Reinoptland commented on src/Pages/HomePage.js:
```diff
@@ -0,0 +1,68 @@
+              <div className="company-logo">
+                <img src={agency.eguideImageSrc} />
+              </div>
+              <div className="agency-card-info" key={agency.id}>
```
This can be turned into a component

Reinoptland reviewed (changes requested):
The styling is much better, but the screen space is not used well on desktop for instance. 

<img width="1095" alt="Screenshot 2022-01-19 at 11 36 05" src="https://user-images.githubusercontent.com/20372832/150114881-fa085b55-919b-4378-a6a4-e0cecc0d44f3.png">

jodelajo committed:
remove comments in css

DariaKryvos committed:
add alt to img

Reinoptland commented:
Incorporate the feedback and merge when you both feel good about it @DariaKryv & @jodelajo

DariaKryvos committed:
added class name to span and link elements on hover  target blank,

jodelajo committed:
Create agency component

DariaKryvos committed:
added style fot decktop version

jodelajo committed:
trying to style

DariaKryvos committed:
changed logo position

DariaKryvos commented:
## What is in this feature
- added alt to img tag
- created component 
- made responsive  design 
- adjusted style

![Screen Shot 2022-01-19 at 3 15 22 PM](https://user-images.githubusercontent.com/47646986/150148894-07407f75-2e49-4d77-8b37-7ad9f60e2665.png)
![Screen Shot 2022-01-19 at 3 16 34 PM](https://user-images.githubusercontent.com/47646986/150148899-666306ca-a6cc-4180-b854-eae43ca923a4.png)
